bot/exts/events/trivianight/_questions.py

```diff
@@ -0,0 +1,210 @@
+            ) / len(all_players)
+            for answer_choice in "ABCD"
+        }
+
+        answers_chosen = dict(sorted(answers_chosen.items(), key=lambda item: item[1], reverse=True))
+
+        answer_embed = Embed(
+            title=f"The correct answer for Question {self.current_question['number']} was..",
+            description=self.current_question["correct"],
+            color=Colours.soft_green
+        )
+
+        for answer, percent in answers_chosen.items():
+            # The `ord` function is used here to change the letter, say 'A' to its corresponding position in the answers
+            answer_embed.add_field(
+                name=f"{percent * 100:.1f}% of players chose",
+                value=self.current_question['answers'][ord(answer) - 65],
+                inline=False
+            )
+
+        self.users_picked = {}
+
+        for button in self.buttons:
+            button.users_picked = self.users_picked
+
+        return return_dict, answer_embed
+
+
+class Questions:
+    """An interface to use from the TriviaNight cog for questions."""
+
+    def __init__(self, scoreboard: Scoreboard):
+        self.scoreboard = scoreboard
+        self.questions = []
+
+    def set_questions(self, questions: list) -> None:
+        """Setting `self.questions` dynamically via a function to set it."""
+        self.questions = questions
+
+    def next_question(self, number: int = None) -> Union[Embed, None]:
+        """
+        Chooses a random unvisited question from the question bank.
+
+        If the number parameter is specified, it'll head to that specific question.
+        """
+        if all("visited" in question.keys() for question in self.questions):
+            return Embed(
+                title=choice(NEGATIVE_REPLIES),
+                description="All of the questions in the question bank have been used.",
+                color=Colours.soft_red
+            )
+
+        if number is None:
+            question_number = randrange(0, len(self.questions))
+            while "visited" in self.questions[question_number].keys():
+                question_number = randrange(0, len(self.questions))
+        else:
+            question_number = number
+
+        self.questions[question_number]["visited"] = True
+        self.view.current_question = self.questions[question_number]
+
+    def list_questions(self) -> str:
+        """
+        Lists all questions from the question bank.
+
+        It will put the following into a message:
+            - Question number
+            - Question description
+            - If the question was already 'visited' (displayed)
+        """
+        spaces = len(sorted(self.questions, key=lambda question: len(question['description']))[-1]["description"]) + 3
+        formatted_string = ""
+        for question in self.questions:
+            formatted_string += f"`Q{question['number']}: {question['description']!r}" \
```

Comment:
This should not be a repr, since this is being displayed to the end user. It outputs something fairly unreadable.
![image](https://user-images.githubusercontent.com/20641196/140191525-e71f7147-45ca-4af7-950d-b6803d7bf462.png)